lib/enumerables.rb

```diff
@@ -0,0 +1,164 @@
+      ans = yield num
+      temp_arr.push(num) if ans == true
+    end
+    temp_arr
+  end
+
+  # my_all
+  def my_all?(paramet = nil)
+    arr = self
+    return true if arr.empty?
+
+    if paramet.nil? && block_given?
+      arr.my_each do |n|
+        ans = yield n
+        return false if ans == false
+      end
+    elsif paramet.nil? && !block_given?
+      arr.my_each { |n| return false if n.nil? || !n }
+    elsif paramet.is_a?(Regexp)
+      arr.my_each { |x| return false unless x.match(paramet) }
+    elsif paramet.is_a?(Module)
+      my_each { |x| return false if x.is_a?(paramet) }
+    else
+      my_each { |x| return false if x != paramet }
+    end
+    true
+  end
+
+  # my_any
+  def my_any?(paramet = nil)
+    arr = self
+    return true if arr.empty?
+
+    if paramet.nil? && block_given?
+      arr.my_each do |n|
+        ans = yield n
+        return true if ans == true
+      end
+    elsif paramet.nil? && !block_given?
+      arr.my_each { |n| return true if !!n == true && !n.nil? }
+    elsif paramet.is_a?(Regexp)
+      arr.my_each { |x| return true if x.match(paramet) }
+    elsif paramet.is_a?(Module)
+      my_each { |x| return true if x.is_a?(paramet) }
+    else
+      my_each { |x| return true if x == paramet }
+    end
+    false
+  end
+
+  # my_none
+  def my_none?(paramet = nil)
+    arr = self
+    return true if arr.empty?
+
+    if paramet.nil? && block_given?
+      arr.my_each do |n|
+        ans = yield n
+        return false if ans == true
+      end
+    elsif paramet.nil? && !block_given?
+      arr.my_each { |n| return false if !!n == true && !n.nil? }
+    elsif paramet.is_a?(Regexp)
+      arr.my_each { |x| return false if x.match(paramet) }
+    elsif paramet.is_a?(Module)
+      my_each { |x| return false if x.is_a?(paramet) }
+    end
+    true
+  end
+
+  # my_count
+  def my_count(para = nil)
+    arr = self
+    if para.nil? && !block_given?
+      counter = 0
+      arr.my_count { counter += 1 }
+      counter
+    elsif !para.nil? && !block_given?
+      counter = 0
+      arr.my_each { |x| counter += 1 if x == para }
+      counter
+    elsif para.nil? && block_given?
+      counter = 0
+      arr.my_each { |x| counter += 1 if yield x }
+      counter
+    end
+  end
+
+  # my_map
+  def my_map(proc = nil)
+    return to_enum unless block_given? || proc
+
+    arr = self
+    new_arr = []
+    if proc
+      arr.my_each { |num| new_arr.push(proc.call(num)) }
+    else
+      arr.my_each { |num| new_arr.push(yield(num)) }
+    end
+    new_arr
+  end
+
+  # my_inject
+  def my_inject(startt = nil)
+    arr = self
+    if startt.nil? && block_given?
+      i = 1
+      sum = arr[0]
+      while i < arr.length
+        sum = yield sum, arr[i]
+        i += 1
+      end
+      sum
+    elsif !startt.nil? && block_given?
+      i = 0
+      sum = startt
+      while i < arr.length
+        sum = yield sum, arr[i]
+        i += 1
+      end
+      sum
+    else
+      0
+    end
+  end
+end
```

Comment:
And finally, these are the problems with `#my_inject`:

```
 18) enumerables #my_inject raises a "LocalJumpError" when no block or argument is given
      Failure/Error: expect { array.my_inject }.to raise_error(LocalJumpError)
        expected LocalJumpError but nothing was raised
      # ./spec/enumerables_spec.rb:406:in `block (3 levels) in <top (required)>'

  19) enumerables #my_inject when a block is given without an initail value combines all elements of enum by applying a binary operation, specified by a block::range
      Failure/Error: sum = arr[0]
      
      NoMethodError:
        undefined method `[]' for 5..50:Range
      # ./lib/enumerables.rb:140:in `my_inject'
      # ./spec/enumerables_spec.rb:424:in `block (4 levels) in <top (required)>'

  20) enumerables #my_inject when a block is given with an initail value combines all elements of enum by applying a binary operation, specified by a block
      Failure/Error: while i < arr.length
      
      NoMethodError:
        undefined method `length' for 5..50:Range
      # ./lib/enumerables.rb:149:in `my_inject'
      # ./spec/enumerables_spec.rb:432:in `block (4 levels) in <top (required)>'

  21) enumerables #my_inject when a symbol is specified without an initail value combines each element of the collection by applying the symbol as a named method
      Failure/Error: expect(array.my_inject(:+)).to eq array.inject(:+)
      
        expected: 378
             got: 0
      
        (compared using ==)
      # ./spec/enumerables_spec.rb:440:in `block (4 levels) in <top (required)>'

  22) enumerables #my_inject when a symbol is specified without an initail value combines each element of the collection by applying the symbol as a named method:range
      Failure/Error: expect(range.my_inject(:*)).to eq range.inject(:*)
      
        expected: 1267253883404724085150525340252698701849068398706688000000000000
             got: 0
      
        (compared using ==)
      # ./spec/enumerables_spec.rb:444:in `block (4 levels) in <top (required)>'

  23) enumerables #my_inject when a symbol is specified with an initail value combines each element of the collection by applying the symbol as a named method
      Failure/Error:
        def my_inject(startt = nil)
          arr = self
          if startt.nil? && block_given?
            i = 1
            sum = arr[0]
            while i < arr.length
              sum = yield sum, arr[i]
              i += 1
            end
            sum
      
      ArgumentError:
        wrong number of arguments (given 2, expected 0..1)
      # ./lib/enumerables.rb:136:in `my_inject'
      # ./spec/enumerables_spec.rb:450:in `block (4 levels) in <top (required)>'

  24) enumerables #my_inject when a symbol is specified with an initail value combines each element of the collection by applying the symbol as a named method::range
      Failure/Error:
        def my_inject(startt = nil)
          arr = self
          if startt.nil? && block_given?
            i = 1
            sum = arr[0]
            while i < arr.length
              sum = yield sum, arr[i]
              i += 1
            end
            sum
      
      ArgumentError:
        wrong number of arguments (given 2, expected 0..1)
      # ./lib/enumerables.rb:136:in `my_inject'
      # ./spec/enumerables_spec.rb:454:in `block (4 levels) in <top (required)>'
```

There are two main problems here:

1. You must prepare your `#my_inject` to work with ranges, just as mentioned with the rest of the methods before.

2. You must prepare `#my_inject` to handle symbols in it's input, like this:

```
(1..3).my_inject(:+)
# This should equal 1+2+3 = 6

(1..3).my_inject(10, :+)
# This should equal 10+1+2+3 = 16
```